xwiki-platform-core/xwiki-platform-url/xwiki-platform-url-default/src/main/java/org/xwiki/url/internal/DefaultURLSecurityManager.java

```diff
@@ -59,6 +59,8 @@
 public class DefaultURLSecurityManager implements URLSecurityManager
 {
     private static final char DOT = '.';
+    private static final char PERCENT = '%';
+    private static final String PERCENT_ESCAPE = "__XWIKI_PERCENT__";
```

Comment:
I wonder if it's okay to assume that this pattern won't be in any URL parameter as the URL can contain user input. What would be safe, imho, is to have a pattern with, e.g., `0__` at the end and to increment that `0` until the pattern isn't part of the input anymore.

Reply:
Frankly the chance that this pattern is used somewhere are very low. I'm really not in favor of adding another layer of complexity here. Now we should probably have some documentation in the API about that escape String, and it should be part of the API.